src/Runner.Worker/ActionCommandManager.cs

```diff
@@ -183,6 +183,33 @@ public sealed class SetEnvCommandExtension : RunnerService, IActionCommandExtens
 
         public void ProcessCommand(IExecutionContext context, string line, ActionCommand command, ContainerInfo container)
         {
+            var serverUrl = context.GetGitHubContext("server_url");
+            var isNotGithub = false;
+            if(!String.IsNullOrWhiteSpace(serverUrl) && serverUrl != "https://github.com")
+            {
+                isNotGithub = true;
+            }
+            
+            var allowUnsecureCommands = false;
+            bool.TryParse(Environment.GetEnvironmentVariable(Constants.Variables.Actions.AllowUnsupportedCommands), out allowUnsecureCommands);
```

Comment:
Be careful, i think bool.TryParse can null ref. You might want to coalesce with empty string.

```
bool.TryParse(Environment.GetEnvironmentVariable(Constants.Variables.Actions.AllowUnsupportedCommands) ?? string.Empty, out allowUnsecureCommands);
```

Reply:
It doesn't, I tested manually and from the docs

```
        //     When this method returns, if the conversion succeeded, contains true if value
        //     is equal to System.Boolean.TrueString or false if value is equal to System.Boolean.FalseString.
        //     If the conversion failed, contains false. The conversion fails if value is null
        //     or is not equal to the value of either the System.Boolean.TrueString or System.Boolean.FalseString
        //     field.
```